api/v1alpha1/hardwareclassificationcontroller_types.go

```diff
@@ -27,14 +27,72 @@ type HardwareClassificationControllerSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of HardwareClassificationController. Edit HardwareClassificationController_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// Namespace under which BareMetalHosts are present
+	Namespace                     string                          `json:"namespace"`
+	ExpectedHardwareConfiguration []ExpectedHardwareConfiguration `json:"expectedValidationConfiguration"`
+}
+
+// ExpectedHardwareConfiguration details to match with the host
+type ExpectedHardwareConfiguration struct {
+	ProfileName string      `json:"profileName"`
+	MinimumCPU  MinimumCPU  `json:"minimumCPU"`
+	MinimumDisk MinimumDisk `json:"minimumDisk"`
+	MinimumNICS MinimumNICS `json:"minimumNICS"`
+	MinimumRAM  int         `json:"minimumRAM"`
```

Comment:
In that we should define default params and it's size if one sends it as empty. 
I will ping you on slack today.

Reply:
what I was proposing is to not check the values not given by the user. So that if they define only a minimum amount of RAM, they are not forced to also define a minimum number of CPU, and the CPU is just not checked when matching the profile. That allows better versatility in the profiles.